src/osdc/ObjectCacher.cc

```diff
@@ -1728,7 +1728,7 @@ int ObjectCacher::writex(OSDWrite *wr, ObjectSet *oset, Context *onfreespace,
       ldout(cct, 10) << "writex writing " << f_it->first << "~"
 		     << f_it->second << " into " << *bh << " at " << opos
 		     << dendl;
-      uint64_t bhoff = bh->start() - opos;
+      uint64_t bhoff = opos - bh->start();
```

Comment:
This change looks correct. But I think the code block has another bug, it only keeps data before 'opos', but does not keep data after 'opos + frag.length()'

Reply:
I think this "bh" created from map_write was supposed to  only map  the new dirty data from ex_it->buffer_extents. it should not contain other old dirty data. otherwise it would be a bug for map_write ?

Reply:
you are right.   thanks